plugins/content/document/lib/operation_engines/KOperationEngineDocument.php

```diff
@@ -56,10 +64,30 @@ protected static function jsonFormat($arr, $basePath)
 				$docMetadata[$key] = json_decode(json_encode($str), true);
 			}
 			kFile::setFileContent($basePath . self::DOC_METADATA_JSON_NAME, json_encode($docMetadata));
+
+		}
+		catch (Exception $e)
+		{
+			KalturaLog::warning("Fail to create json file: " . $e->getMessage());
+		}
+	}
+
+	protected static function encryptFileName($basePath, $fileName, $key) {
+		if (!$key )
+		{
+			return $fileName;
+		}
+		try {
+			$pathinfo = pathinfo ($fileName);
+			$encryptName = base64_encode(hash_hmac('sha256', $pathinfo['filename'], $key, true));
```

Comment:
i think we better make sure is really set before accessing $pathinfo['filename'] just in case, same for extension

Reply:
Well, this wrap in try-catch block, so if we failing I would expect to not encrypt the files name at all.
I have no objection to add protection if you want. Do you want me to check for exist before passing on?